firedrake/mg/ufl_utils.py

```diff
@@ -384,7 +385,8 @@ def create_interpolation(dmc, dmf):
     mat.setType(mat.Type.PYTHON)
     mat.setPythonContext(ctx)
     mat.setUp()
-    return mat, None
+    rscale = mat.createVecLeft() if row_size == col_size else None
```

Comment:
Not sure wether Right or Left is the correct thing here. But it does not really matter if the sizes are the same.